### PR TITLE
You can now search person by document list and you can call it with rpc

### PR DESCRIPTION
**Pull request by BatrakM**

### CODE REVIEW

File: apps/core/lib/core/rpc/rpc.ex [hunk 1]
```diff
@@ -0,0 +1,10 @@
+defmodule Core.Rpc do
+  @moduledoc false
+
+  alias Core.Persons.PersonsAPI
+
+  @spec search_person(params :: map()) :: {:error, any()} | {:ok, list(map)}
```
Comment (AlexKovalevych) [inline apps/core/lib/core/rpc/rpc.ex]: please add detailed doc about input and output parameters

File: apps/core/lib/core/persons/persons_api.ex [hunk 2]
```diff
@@ -157,12 +158,36 @@ defmodule Core.Persons.PersonsAPI do
...
+    query =
+      join(query,
+        :inner,
+        [p],
+        d in PersonDocument,
+        d.person_id == p.id
+      )
+
+    documents_query = dynamic([p, d], d.type == ^document["type"] and fragment("lower(?) = lower(?)", d.number, ^document["number"]))
+    documents_query = Enum.reduce(documents, documents_query, fn document, acc -> dynamic([p, d], ^acc or d.type == ^document["type"] and fragment("lower(?) = lower(?)", d.number, ^document["number"])) end)
```
Comment (AlexKovalevych) [inline apps/core/lib/core/persons/persons_api.ex]: please format long line to something more readable

File: apps/core/test/unit/rpc/rpc_test.exs [hunk 3]
```diff
@@ -0,0 +1,150 @@
+defmodule Core.RpcTest do
+  @moduledoc false
+
+  use Core.ModelCase, async: true
+  import Core.Factory
+  # alias Core.PRMRepo
```
Comment (AlexKovalevych) [inline apps/core/test/unit/rpc/rpc_test.exs]: please remove comments

Comment (BatrakM) [reply apps/core/test/unit/rpc/rpc_test.exs]: done

File: apps/core/lib/core/persons/persons_api.ex (hunk 2 above)
Comment (BatrakM) [reply apps/core/lib/core/persons/persons_api.ex]: done

File: apps/core/lib/core/rpc/rpc.ex (hunk 1 above)
Comment (BatrakM) [reply apps/core/lib/core/rpc/rpc.ex]: Done

File: apps/core/lib/core/persons/persons_api.ex [hunk 4]
```diff
@@ -157,12 +158,39 @@ defmodule Core.Persons.PersonsAPI do
...
+
+    query
+    |> from()
+    |> where(^documents_query)
+    |> distinct(true)
+  end
+
+  defp with_documents(query, _), do: query
+
+  defp document_search_query(type, number), do: dynamic([p, d], d.type == ^type and fragment("lower(?) = lower(?)", d.number, ^number))
```
Comment (AlexKovalevych) [inline apps/core/lib/core/persons/persons_api.ex]: please use multiline syntax for such functions

Comment (BatrakM) [reply apps/core/lib/core/persons/persons_api.ex]: done

File: apps/core/lib/core/persons/persons_api.ex [hunk 5]
```diff
@@ -157,12 +158,41 @@ defmodule Core.Persons.PersonsAPI do
...
+      d.person_id == p.id and d.type == ^type and fragment("lower(?) = lower(?)", d.number, ^number)
     )
   end
 
   defp with_type_number(query, _), do: query
 
+  defp with_documents(query, %{"documents" => []}), do: query
+
+  defp with_documents(query, %{"documents" => documents}) do
+    [document | documents] = documents
```
Comment (progsmile) [inline apps/core/lib/core/persons/persons_api.ex]: Can match in function declaration
`defp with_documents(query, %{"documents" => [document | documents]}) do `

File: apps/core/lib/core/persons/persons_api.ex [hunk 6]
```diff
@@ -157,12 +158,41 @@ defmodule Core.Persons.PersonsAPI do
...
-      d.person_id == p.id and d.type == ^type and d.number == ^number
+      d.person_id == p.id and d.type == ^type and fragment("lower(?) = lower(?)", d.number, ^number)
     )
   end
 
   defp with_type_number(query, _), do: query
 
+  defp with_documents(query, %{"documents" => []}), do: query
+
+  defp with_documents(query, %{"documents" => documents}) do
```
Comment (progsmile) [inline apps/core/lib/core/persons/persons_api.ex]:
```suggestion
  defp with_documents(query, %{"documents" => [document | documents]}) do
```

File: apps/core/lib/core/persons/persons_api.ex (hunk 5 above)
Comment (progsmile) [inline apps/core/lib/core/persons/persons_api.ex]:
```suggestion
```

**Pull request merged**